torchrl/envs/vec_env.py

```diff
@@ -359,42 +365,51 @@ def _create_td(self) -> None:
                     ].keys(True, True)
                 )
                 env_output_keys = env_output_keys.union(
-                    unravel_key(("next", key))
+                    key
                     for key in meta_data.specs["output_spec"][
                         "full_observation_spec"
                     ].keys(True, True)
                 )
-            env_output_keys = env_output_keys.union(
-                {
-                    unravel_key(("next", key))
-                    for key in self.reward_keys + self.done_keys
-                }
-            )
-            self.env_obs_keys = sorted(env_obs_keys, key=_sort_keys)
-            self.env_input_keys = sorted(env_input_keys, key=_sort_keys)
-            self.env_output_keys = sorted(env_output_keys, key=_sort_keys)
+            env_output_keys = env_output_keys.union(self.reward_keys + self.done_keys)
+            self._env_obs_keys = sorted(env_obs_keys, key=_sort_keys)
+            self._env_input_keys = sorted(env_input_keys, key=_sort_keys)
+            self._env_output_keys = sorted(env_output_keys, key=_sort_keys)
 
         self._selected_keys = (
-            set(self.env_output_keys)
-            .union(self.env_input_keys)
-            .union(self.env_obs_keys)
+            set(self._env_output_keys)
+            .union(self._env_input_keys)
+            .union(self._env_obs_keys)
             .union(set(self.done_keys))
         )
         self._selected_keys.add("_reset")
 
-        self._selected_reset_keys = self.env_obs_keys + self.done_keys
-        self._selected_step_keys = self.env_output_keys
+        # input keys
+        self._selected_input_keys = {
+            _unravel_key_to_tuple(key) for key in self._env_input_keys
+        }
+        # output keys after reset
+        self._selected_reset_keys = {
+            _unravel_key_to_tuple(key)
+            for key in self._env_obs_keys + self.done_keys + ["_reset"]
```

Comment:
reset keys should not include "_reset" right? 
if so there will be multiple _reset keys (one per done)

Reply:
We need the "_reset" keys in the reset_keys as these need to be synced across processes